internal/node/model/models.go

```diff
@@ -0,0 +1,68 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package model
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type Hash = common.Hash
+type Address = common.Address
+type InputCompletionStatus string
+type ClaimStatus string
+
+const (
+	InputStatusNone                       InputCompletionStatus = "NONE"
+	InputStatusAccepted                   InputCompletionStatus = "ACCEPTED"
+	InputStatusRejected                   InputCompletionStatus = "REJECTED"
+	InputStatusException                  InputCompletionStatus = "EXCEPTION"
+	InputStatusMachineHalted              InputCompletionStatus = "MACHINE_HALTED"
+	InputStatusCycleLimitExceeded         InputCompletionStatus = "CYCLE_LIMIT_EXCEEDED"
+	InputStatusTimeLimitExceeded          InputCompletionStatus = "TIME_LIMIT_EXCEEDED"
+	InputStatusPayloadLengthLimitExceeded InputCompletionStatus = "PAYLOAD_LENGTH_LIMIT_EXCEEDED"
+)
+
+const (
+	ClaimStatusPending   ClaimStatus = "PENDING"
+	ClaimStatusSubmitted ClaimStatus = "SUBMITTED"
+	InputStatusFinalized ClaimStatus = "FINALIZED"
+)
+
+type Application struct {
+	Id                 uint64
+	AppAddress         Address
+	TemplateHash       Hash
+	SnapshotURI        string
+	LastProcessedBlock uint64
+	EpochLength        uint64
+}
+
+type Input struct {
+	Id               uint64
+	Index            uint64
+	CompletionStatus InputCompletionStatus
+	Blob             hexutil.Bytes
+	BlockNumber      uint64
+	MachineHash      Hash
+	OutputsHash      Hash
+	AppAddress       Address
+}
+
+type Output struct {
+	Id                   uint64
+	Index                uint64
+	Blob                 hexutil.Bytes
+	OutputHashesSiblings []Hash
+	InputId              uint64
+}
+
+type Claim struct {
+	Id                   uint64
+	Status               ClaimStatus
+	FirstBlock           uint64
+	LastBlock            uint64
+	OutputMerkleRootHash Hash
+	AppAddress           Address
+}
```

Comment:
Report and output share the same base values, as reports are outputs that are not verifiable. We differentiate them on the database to facilitate cataloguing, but since they use the same columns, I think there's no need to have another model.

Reply:
I don't think it is good to have a `Report` be an `Output` with missing fields.
We will be using that in other parts of the code, and it can generate confusion.
`Report` being a type also adds type safety to operations.

Reply:
In what parts of the code will we use reports? From what I understand, once we put them on the database, they are only used in the GraphQL for inspect purposes.

The `machine-advancer` is the only service of the Node where this will come into play, because it will get all the outputs and needs to separate reports and outputs to properly insert them in the database. Since you're responsible for it, if you think it will help, I can add it, no problem.

Reply:
There's also the critical distinction that a `Report` is not provable, so it does make sense to make it its own, separate struct to avoid confusion.

Reply:
The Inspect Server inside the HTTP Server will also use `Report`.